### ISSUE TITLE
About the question that train loss is greater than valid loss and greater than test loss after using my own dataset

### ISSUE THREAD

**Issue by sunxiaoyao-git** [LarsBentsen/FFTransformer#7]
Hello!
After reading and debugging your code, I brought in my own dataset and found that the loss curve had a large error in train, but much smaller error in valid and test. What might be the cause of this? Does it depend on the distribution of the original data? 

I tried univariate prediction and not used graph network.
![image](https://github.com/LarsBentsen/FFTransformer/assets/114201981/9184b69a-7627-4458-886c-ad2b1957eedd)

**Comment by LarsBentsen** [comment]
Hi! 
These are fairly common characteristics that you might observe when training deep learning models. There are a host of different possible reasons, and I've outlined some immediate thoughts below: 
- dropout: If you're using dropout, then this will be turned off when you validate and might be a reason for seeing higher training losses (i.e. when dropout is on) than validation losses. If this is the reason, it is not something to worry about and is something that you might expect, at least in the earlier training stages.
- Another reason could be that there are slight differences in the data distributions of the training, test and validation sets and that the scenarios encountered in the validation set are easier to predict than those in the training set. This is also fairly common as the training dataset is typically much larger than the test and validation sets and might therefore contain slightly more diverse and challenging examples. However, if there are distribution shifts from the training to validation sets, you might also observe the opposite characteristics, with higher validation losses than training losses. 
- Finally, this was only after 6 epochs of training. You could also try to train the models for longer to see if you still observe the same characteristics for longer-trained models. 
- There are a range of other potential explanations and I'm sure there will be a large number of similar questions posted on StackExchange with more thorough explanations and discussions. 

However, it should be noted that this is not necessarily an _issue_ nor mean that anything is wrong with the model. 
As a second note, I would also like to point out that you typically do not compute test losses during training or when selecting models as this can be considered "cheating". This is because choosing models based on the test set means that the test set no longer represents a completely new and unseen scenario.  Usually, the test set should be held out until testing the model (i.e. when all training, model selection and etc. is finished). 

I hope this answered some of your questions and I will close the issue now, but feel free to re-open if you still require some additional clarifications/discussion. Also, I'm glad that you've decided to test our models yourself and please remember to cite the paper if you found it useful for your own research! :-)
